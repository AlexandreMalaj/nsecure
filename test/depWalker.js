"use strict";

// Require Node.js Dependencies
const { join } = require("path");

// Require Internal Dependencies
const { depWalker } = require("../src/depWalker");

// CONSTANTS
const FIXTURE_PATH = join(__dirname, "fixtures/depWalker");

// JSON PAYLOADS
const is = require(join(FIXTURE_PATH, "slimio.is.json"));

test("should return null for depWalker of @slimio/is", async() => {
    const result = await depWalker(is, { verbose: false });
    const resultAsJSON = JSON.parse(JSON.stringify(Object.fromEntries(result), null, 2));
    delete resultAsJSON["@slimio/is"]["1.5.1"].size;
    expect(resultAsJSON).toMatchSnapshot();
});
