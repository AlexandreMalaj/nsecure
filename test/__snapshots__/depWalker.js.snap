// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return null for depWalker of @slimio/is 1`] = `
Object {
  "@slimio/is": Object {
    "1.5.1": Object {
      "author": "SlimIO",
      "composition": Object {
        "extensions": Array [
          "",
          ".js",
          ".json",
          ".md",
          ".toml",
          ".ts",
        ],
        "minified": Array [],
        "required": Array [
          "./src/utils",
        ],
        "required_builtin": Array [],
        "unusedOrMissing": Array [],
      },
      "description": "SlimIO is (JavaScript Primitives &amp; Objects type checker)",
      "flags": Object {
        "hasCustomResolver": false,
        "hasDependencies": false,
        "hasExternalCapacity": false,
        "hasIndirectDependencies": false,
        "hasLicense": true,
        "hasManifest": true,
        "hasMinifiedCode": false,
        "hasMissingOrUnusedDependency": false,
        "hasMultipleLicenses": false,
        "hasScript": false,
        "hasSuspectImport": false,
        "isDeprecated": false,
        "isGit": false,
      },
      "gitUrl": null,
      "id": 0,
      "license": Object {
        "hasMultipleLicenses": false,
        "licenses": Array [
          Object {
            "from": "package.json",
            "spdx": Object {
              "fsf": true,
              "fsfAndOsi": true,
              "includesDeprecated": false,
              "osi": true,
            },
            "spdxLicenseLinks": Array [
              "https://spdx.org/licenses/MIT.html#licenseText",
            ],
            "uniqueLicenseIds": Array [
              "MIT",
            ],
          },
          Object {
            "from": "LICENSE",
            "spdx": Object {
              "fsf": true,
              "fsfAndOsi": true,
              "includesDeprecated": false,
              "osi": true,
            },
            "spdxLicenseLinks": Array [
              "https://spdx.org/licenses/MIT.html#licenseText",
            ],
            "uniqueLicenseIds": Array [
              "MIT",
            ],
          },
        ],
        "uniqueLicenseIds": Array [
          "MIT",
        ],
      },
      "usedBy": Object {},
    },
    "metadata": Object {
      "author": "SlimIO",
      "authors": Array [],
      "hasChangedAuthor": false,
      "hasManyPublishers": false,
      "homepage": "https://github.com/SlimIO/is#readme",
      "lastUpdateAt": "2019-06-11T06:41:41.590Z",
      "lastVersion": "1.5.1",
      "publishedCount": 7,
      "publishers": Array [
        Object {
          "firstPublishAt": "2018-09-22T20:36:27.523Z",
          "name": "fraxken",
          "version": "1.0.0",
        },
      ],
    },
    "versions": Array [
      "1.5.1",
    ],
    "vulnerabilities": Array [],
  },
}
`;
